src/App.tsx

```diff
@@ -17,18 +62,34 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                query={query}
+                setQuery={setQuery}
+                setTaskStatusFilter={setTaskStatusFilter}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {isTodoArrayLoaded ? (
+                <TodoList
+                  todos={filteredTodos}
+                  selectedTodo={selectedTodo}
+                  selectTodo={setSelectedTodo}
+                />
+              ) : (
+                <Loader />
+              )}
             </div>
           </div>
         </div>
       </div>
 
-      <TodoModal />
+      {selectedTodo !== null && (
```

Comment:
```suggestion
      {selectedTodo && (
```
You don't need to compare with null, https://developer.mozilla.org/en-US/docs/Glossary/Truthy